src/player.cpp

```diff
@@ -4319,6 +4360,57 @@ void player::regen( int rate_multiplier )
         }
     }
 
+    // include healing effects
+    for( int i = 0; i < num_hp_parts; i++ ) {
```

Comment:
Consider iterating over all body parts via something like `for( const body_part bp : all_body_parts ) { ...`

Note that `body_part` is implicitly convertible to an integer, so you can use it directly as index into an array (e.g. no separate `int` variable needed).

Reply:
But `all_body_parts` includes feets and hands, so it can not be used here directly.